packages/react/ds/src/alert/alert.tsx

```diff
@@ -0,0 +1,103 @@
+'use client';
+import { useState, type ReactNode } from 'react';
+import { tv, type VariantProps } from 'tailwind-variants';
+import { cn } from '../cn.js';
+import { Icon } from '../icon/icon.js';
+import { IconButton } from '../icon-button/icon-button.js';
+
+const alertVariants = tv({
+  slots: {
+    base: 'gi-alert-base',
+    container: 'gi-alert-container',
+    heading: 'gi-alert-title',
+    dismiss: 'gi-alert-dismiss',
+  },
+  variants: {
+    variant: {
+      info: {
+        base: 'gi-alert-info',
+      },
+      danger: {
+        base: 'gi-alert-danger',
+      },
+      success: {
+        base: 'gi-alert-success',
+      },
+      warning: {
+        base: 'gi-alert-warning',
+      },
+    },
+  },
+  defaultVariants: {
+    variant: 'info',
+  },
+});
+
+type AlertProps = {
+  variant?: VariantProps<typeof alertVariants>['variant'];
+  title: string;
+  children?: ReactNode;
+  className?: string;
+  dismissible?: boolean;
+};
+
+const icon = ({ variant }: VariantProps<typeof alertVariants>) => {
+  let icon;
+  switch (variant) {
+    case 'warning': {
+      icon = 'warning';
+      break;
+    }
+    case 'success': {
+      icon = 'check_circle';
+      break;
+    }
+    case 'danger': {
+      icon = 'error';
+      break;
+    }
+    default: {
+      icon = 'info';
+    }
+  }
+  return icon;
+};
+
+function Alert({
+  title,
+  children,
+  variant = 'info',
+  dismissible,
+  className,
+}: AlertProps) {
+  const [isDismissed, setIsDismissed] = useState(false);
+
+  const { base, heading, container, dismiss } = alertVariants({
+    variant,
+  });
+
+  if (isDismissed) {
+    return null;
+  }
+  return (
+    <div className={cn(base(), className)} role="alert">
+      <Icon icon={icon({ variant })} />
+      <div className={container()}>
+        <p className={heading()}>{title}</p>
+        {children}
+      </div>
+      {dismissible && (
+        <IconButton
+          onClick={() => setIsDismissed(true)}
```

Comment:
maybe adding an "onClick" handler as prop for when the alert is closed it would allow the parent component to detect the closure and perform some action

Reply:
Don't know about adding the `onClick` as a prop in this case since the action is intended just for closing the `Alert`. Not sure I see a different use case other than that. Thoughts @pregno @hamza14khan  ?

Reply:
Good point. 
We can have an optional callback props `onClose` (or `onDismiss`) that is triggered when the user press the `X` close button.

Reply:
Added